packages/compass-preferences-model/src/renderer-ipc.ts

```diff
@@ -14,13 +14,13 @@ import { createSandboxAccessFromProps } from './setup-preferences';
  * API to communicate with preferences from the electron renderer process.
  */
 export const makePreferencesIpc = (ipcRenderer: HadronIpcRenderer) => {
-  const cachedPreferences = {} as AllPreferences;
+  let cachedPreferences = {} as AllPreferences;
+  let inflightCacheRefresh: Promise<AllPreferences> | undefined;
   async function refreshCachedPreferences(): Promise<AllPreferences> {
-    const result: AllPreferences = await ipcRenderer.invoke(
-      'compass:get-preferences'
-    );
-    Object.assign(cachedPreferences, result);
```

Comment:
(drive-by change, there’s no reason not to just fully re-assign the variable; `Object.assign` would not delete keys that no longer exist, which currently does not happen but might at some point in the future)